addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -203,6 +203,70 @@ def action_assign(self):
         self.ensure_one()
         self.picking_ids.action_assign()
 
+    def action_put_in_pack(self):
+        """ Action to put move lines with 'Done' quantities into a new pack
+        This method follows same logic to stock.picking.
+        """
+        self.ensure_one()
+        if self.state not in ('done', 'cancel'):
+            picking_move_lines = self.move_line_ids
+
+            move_line_ids = picking_move_lines.filtered(lambda ml:
+                float_compare(ml.qty_done, 0.0, precision_rounding=ml.product_uom_id.rounding) > 0
+                and not ml.result_package_id
+            )
+            if not move_line_ids:
+                move_line_ids = picking_move_lines.filtered(lambda ml: float_compare(ml.product_uom_qty, 0.0,
+                                     precision_rounding=ml.product_uom_id.rounding) > 0 and float_compare(ml.qty_done, 0.0,
+                                     precision_rounding=ml.product_uom_id.rounding) == 0)
+            if move_line_ids:
+                res = self.picking_ids[0]._pre_put_in_pack_hook(move_line_ids)
+                if not res:
+                    res = self._put_in_pack(move_line_ids)
+                return res
+            else:
+                raise UserError(_("Please add 'Done' quantities to the batch picking to create a new pack."))
+
+    # -------------------------------------------------------------------------
+    # Put in pack methods
+    # -------------------------------------------------------------------------
+
+    def _put_in_pack(self, move_line_ids):
```

Comment:
Instead of duplicate the code for this method, will be possible to change the `_put_in_pack` in stock and add a optional parameter `create_level_package=True` by example. As you say it is quite sad to duplicate and do the maintenance of this part twice.

Reply:
Good point, I made this change.